bluemira/equilibria/opt_constraints.py

```diff
@@ -61,6 +62,44 @@ def _get_dummy_equilibrium(equilibrium):
     return equilibrium
 
 
+def spherical_harmonics_constraint(
+    constraint, vector, grad, eq, r_t, initial_harmonics, max_degree
+):
+    """
+    Constraint function to constrain spherical harmonics starting from initial
+    coil currents and associated core plasma.
+
+    Parameters
+    ----------
+    eq: Equilibrium
+        Equilibrium used to for coilset.
+    r_t: float
+        Typical length scale of the problem (e.g. radius at outer midplane)
+    initial_harmonics: np.ndarry
+        Initial harmonic amplitudes obtained from desired core plasma
+    max_degree: float
+        Maximum degree of spherical harmonics desired to constrain.
+    """
+    # Does the eq get updated at each iteration, requiring I_f as input or can
+    # we just extract I_f from the eq object within the function?
```

Comment:
The equilibrium gets update at each grad-Shafranov iteration by definition.

As far as I understand the GS does not get updated with the PF optimization, however the coilset by definition should. Although I am not fully aware of the implentation details @alexanderianblair ?